flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/OperationConverterUtils.java

```diff
@@ -117,6 +128,193 @@ public static Operation convertAddReplaceColumns(
+            SqlAlterTableDropColumns sqlAlterTableDropColumns) {
+        Schema originSchema = catalogTable.getUnresolvedSchema();
+        List<String> originTableColumns =
+                originSchema.getColumns().stream()
+                        .map(Schema.UnresolvedColumn::getName)
+                        .collect(Collectors.toList());
+
+        // filter the dropped column which is not in table firstly
+        List<String> toDropColumns =
+                sqlAlterTableDropColumns.getColumns().getList().stream()
+                        .map(SqlIdentifier.class::cast)
+                        .map(SqlIdentifier::getSimple)
+                        .filter(column -> originTableColumns.contains(column))
+                        .collect(Collectors.toList());
+
+        // validate column size
+        if (originTableColumns.size() == 1 && toDropColumns.size() > 0) {
+            throw new ValidationException(
+                    String.format(
+                            "Table %s only has one column, please use DROP TABLE syntax.",
+                            tableIdentifier));
+        }
+
+        // validate the dropped column is referenced by computed column
+        toDropColumns.forEach(
+                column -> validateComputedColumn(column, originTableColumns, originResolveSchema));
+
+        // validate the dropped column is referenced by watermark
+        toDropColumns.forEach(
+                column ->
+                        validateWatermark(
+                                column,
+                                originResolveSchema.getWatermarkSpecs(),
+                                originTableColumns));
+
+        Schema.Builder builder = Schema.newBuilder();
+        // build column
+        builder.fromColumns(
+                originSchema.getColumns().stream()
+                        .filter(originColumn -> !toDropColumns.contains(originColumn.getName()))
+                        .collect(Collectors.toList()));
+
+        // build watermark
+        originSchema
+                .getWatermarkSpecs()
+                .forEach(
+                        watermarkSpec ->
+                                builder.watermark(
+                                        watermarkSpec.getColumnName(),
+                                        watermarkSpec.getWatermarkExpression()));
+        // build primary key
+        Optional<Schema.UnresolvedPrimaryKey> originPrimaryKey = originSchema.getPrimaryKey();
+        if (originPrimaryKey.isPresent()) {
+            List<String> originPrimaryKeyNames = originPrimaryKey.get().getColumnNames();
+            String constrainName = originPrimaryKey.get().getConstraintName();
+            List<String> newPrimaryKeyNames =
+                    originPrimaryKeyNames.stream()
+                            .filter(pkName -> !toDropColumns.contains(pkName))
+                            .collect(Collectors.toList());
+            if (newPrimaryKeyNames.size() > 0) {
```

Comment:
What the behavior of drop one column of composite primary key in other databases?

Reply:
mysql and postgresql.Here is some scenes when drop column:

Constraint refers to this column and other column.Remove this column from Constraint.If constraint only refers to this column ,drop column and constraint together.